### ISSUE TITLE
Running on SICK dataset

### ISSUE THREAD

**Issue by kurtespinosa** [sapruash/RecursiveNN#1]
Hi, thank you for sharing this code! I wonder if you could point out/explain a bit which part of the code I would need to change to use the SICK dataset?  Thank you so much.

Kurt

**Comment by sapruash** [comment]
Hi Kurt,
You would need to make some modifications to the code to  make it work for semantic relatedness task. 
Since, you would be working with pairsof sentences, change input placeholders to handle an additional input sentence (the second of the pair).
Calculate leaf embeddings for both sentences (call it one after another). 
Calculate root embedding for both sentences (call compute state function here)
Concatenate the multiplicative and subtractive parts of the two root embeddings. (see the original paper).
Rest follow the paper ( you simply have to pass the concatenation through an additional  layer (add it to the add_model_variables before projection).
Do the projection and then compute the cross entropy loss.

**Comment by kurtespinosa** [comment]
Thanks Sapruash for your quick reply. I will update you.
On 7 Feb 2017, at 2:19 am, sapruash <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Kurt,
You would need to make some modifications to the code to make it work for semantic relatedness task.
Since, you would be working with pairsof sentences, change input placeholders to handle an additional input sentence (the second of the pair).
Calculate leaf embeddings for both sentences (call it one after another).
Calculate root embedding for both sentences (call compute state function here)
Concatenate the multiplicative and subtractive parts of the two root embeddings. (see the original paper).
Rest follow the paper ( you simply have to pass the concatenation through an additional layer (add it to the add_model_variables before projection).
Do the projection and then compute the cross entropy loss.

**Comment by martin-andor** [comment]
Hi Kurt,

Did you manage to make the code work for a two-sentence input in the end? I am an NLP master student and would like to implement Sheng Tai's paper in tensorflow 2 for a paraphrase detection task for my final project.
